src/Planners/src/ConvexHullHelper.cpp

```diff
@@ -24,67 +26,106 @@ ConvexHullHelper::ConvexHullHelper()
 {
     m_pimpl = std::make_unique<Impl>();
 
-    m_pimpl->A.resize(0,0);
+    m_pimpl->A.resize(0, 0);
     m_pimpl->b.resize(0);
 }
 
-ConvexHullHelper::~ConvexHullHelper()
-{
-}
+ConvexHullHelper::~ConvexHullHelper() = default;
 
 bool ConvexHullHelper::buildConvexHull(Eigen::Ref<const Eigen::MatrixXd> points)
 {
+    constexpr auto logPrefix = "[ConvexHullHelper::buildConvexHull]";
+
     const std::size_t numberOfPoints = points.cols();
     const std::size_t numberOfCoordinates = points.rows();
 
-    // the qhull object can be called only once
-    orgQhull::Qhull qhull;
+    // required since qh_new_qhull expects a non const pointer to a double
+    Eigen::Matrix<coordT, Eigen::Dynamic, Eigen::Dynamic> pointsCoordinates = points;
+
+    facetT* facet = nullptr;
+    vertexT* vertex = nullptr;
 
-    // it seems that the pointCoordinates element cannot be cleaned so a new point coordinates has to be instantiate
-    orgQhull::PointCoordinates pointCoordinates;
-    pointCoordinates.setDimension(numberOfCoordinates);
+    // Qhull's data structure.
+    qhT qh_qh;
+    qhT* qh = &qh_qh;
 
-    std::vector<double> allCoordinates(numberOfPoints * numberOfCoordinates);
-    Eigen::Map<Eigen::MatrixXd>(allCoordinates.data(), numberOfCoordinates, numberOfPoints) = points;
+    // initialize qhull
+    qh_zero(qh, nullptr);
 
-    // map = points;
-    pointCoordinates.append(allCoordinates);
+    // set the input
+    char flags[250];
+    sprintf(flags, "qhull");
```

Comment:
```suggestion
    sprintf(flags, "qhull ");
```

Hopefully this should fix the runtime behaviour with qhull from 20.04, see https://github.com/PointCloudLibrary/pcl/blob/3eabe5a772717028874db24a54bc6f88bffffae8/surface/include/pcl/surface/convex_hull.h#L131C44-L131C44 .